src/Avalonia.Dialogs/Internal/ManagedFileChooserFilterViewModel.cs

```diff
@@ -19,7 +19,7 @@ public ManagedFileChooserFilterViewModel(FilePickerFileType filter)
             }
 
             _patterns = filter.Patterns?
-                .Select(e => new Regex(Regex.Escape(e).Replace(@"\*", ".*").Replace(@"\?", "."), RegexOptions.Singleline | RegexOptions.IgnoreCase))
+                .Select(e => new Regex(Regex.Escape(e).Replace(@"\*", ".*").Replace(@"\?", "."), RegexOptions.Singleline | RegexOptions.IgnoreCase | RegexOptions.Compiled))
```

Comment:
Won't this mean that the regex is compiled for each element in `filter.Patterns`? From the [docs](https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regexoptions?view=net-7.0):

> Compiled regular expressions maximize run-time performance at the expense of initialization time.

I would have thought that doing that in a loop is probably not the best idea? Especially when by default it will only be ran once. For example looking at the benchmarks here it looks like `Compiled` only shows improvements in speed when doing a lot of matches: https://stackoverflow.com/questions/513412/how-does-regexoptions-compiled-work